### ISSUE TITLE
question about training sequence length

### ISSUE THREAD

**Issue by kibong** [lmsdss/LayerNorm-Scaling#4]
The default is set to sequence length 256
https://github.com/lmsdss/LayerNorm-Scaling/blob/4d5ffda533016ef583d5df6d6a07c07d840872a2/torchrun_main.py#L42
have you tried any experiments with longer sequence lengths?
when I ran it with 4K length, COD seemed worse than Pre-LN from a loss perspective

**Comment by lmsdss** [comment]
Thank you for the feedback!

I did run experiments with longer sequence lengths, specifically with a sequence length of 2K, using a 130M model, batch size of 128, and 20,000 training steps. The results show that, for this setup, COD still outperforms Pre-LN in terms of loss. 

Here are the metrics:

Pre-LN:
Loss: 3.405374885
Perplexity:  30.12558723
COD:
Loss: 3.35370779
Perplexity: 28.60861197

Even with the longer sequence length, COD remains slightly better in terms of loss. That said, I agree that 4K sequences should work as well.  I’d recommend double-checking the code settings and configuration to ensure everything is aligned.

Let me know if you'd like to try different configurations or have any other questions!

**Comment by kibong** [comment]
Thank you for your response :)
I have one more question.
I got the above results by training with packed training data through something like [this](https://huggingface.co/docs/transformers/main_classes/data_collator#transformers.DataCollatorWithFlattening)

Did you also use packing in the experiments you shared? (Because without doing so, C4 samples aren't long enough, so even with 2K or 4K context lengths, most would be padding)

**Comment by kibong** [comment]
When I experimented with the latest Megatron LM, I observed that COD performed better than the baseline!
I think there might have been some issue with my previous HF experimental setup.
I'll close this issue.
Thank you

**Comment by Shiweiliuiiiiiii** [comment]
> When I experimented with the latest Megatron LM, I observed that COD performed better than the baseline! I think there might have been some issue with my previous HF experimental setup. I'll close this issue. Thank you

Hi, Would it be possible for you to share your experimental settings (what model and datasets) and results with us? I'm just curious about it.
